### DISCUSSION TITLE
Volume control ESpeak espeak(i2s);

### DISCUSSION THREAD

**Discussion by pinter75** [pschatzmann/arduino-espeak-ng#12] [Q&A] [answered]
Hi,

I am wondering how I could add dynamic volume control to the i2s stream?

I2SStream i2s; // or replace with AudioKitStream for AudioKit
ESpeak espeak(i2s);

  // add voice option
  espeak.add("/mem/data/voices/!v/benjamin", espeak_ng_data_voices__v_benjamin, espeak_ng_data_voices__v_benjamin_len);
  espeak.setVoice("en+benjamin");
  // setup espeak
  espeak.setRate(80);
  espeak.setWordGap(50);
  espeak.begin();
  // Set voice and voice option

  // setup output
  audio_info espeak_info = espeak.audioInfo();
  auto cfg = i2s.defaultConfig();
  cfg.pin_ws = 25; //D2
  cfg.pin_bck = 26; //D3
  cfg.pin_data = 27; //D4
  cfg.channels = espeak_info.channels; // 1
  cfg.sample_rate = espeak_info.sample_rate; // 22050
  cfg.bits_per_sample = espeak_info.bits_per_sample; // 16
  i2s.begin(cfg);

**Comment by pschatzmann** [comment, marked as answer]
https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control
